src/options.cpp

```diff
@@ -665,8 +667,8 @@ int zmq::options_t::setsockopt (int option_,
 
         case ZMQ_HEARTBEAT_TTL:
             // Convert this to deciseconds from milliseconds
-            value = value / 100;
-            if (is_int && value >= 0 && value <= 6553) {
```

Comment:
I'm not sure this was a typo? Right now the max TTL is one hour and a half, this would make it 18 hours which is a bit much

https://rfc.zeromq.org/spec:37/ZMTP

Reply:
Ok... that's really in the spec... But it is a very strange value, isn't it? ;)

Anyway, I will change it back to the previous value, but it makes sense to add the tests, so I keep the PR open and fix.

Reply:
It's very very strange :-)

The spec is in DRAFT form, so feel free to amend it to something less strange if you want to

Reply:
Oh, I checked it again... and now I understand, but it is even a bit weirder:

First, the spec says:
> The ping-ttl is a 16-bit unsigned integer in network order that MAY be zero or MAY contain a time-to-live *measured in tenths of seconds*

This would allow values up to UINT16_MAX==65535 deciseconds, as my PR implements.

Then it says:
> The maximum TTL is 6553 seconds

This would allow values up to 65530 deciseconds

So UINT16_MAX is almost correct according to the spec, in fact it were UINT16_MAX - UINT16_MAX % 10.

However, I think this second sentence is not meant to restrict the value, but simply explain the consequence of having a 16 bit unsigned integer specifying deciseconds.

Maybe the wording in the spec should be changed to:
> As a consequence, the maximum TTL is 6553.5 seconds

Reply:
right, that makes even more sense, feel free to send a pr to the rfc and I'll merge that too

Reply:
Ok, I will also add this to doc/zmq_setsockopt.txt

Reply:
Done